### PR TITLE
docs: add usage

### LINKED ISSUE #42: How to actually use the captcha on a page?
Documentation is great for setting up, but how do I actually use the recaptcha on a given page? After seting it up, I'm intercepting the request params but there's no captcha params at all.

### LINKED ISSUE #64: is there any docs for this module?
Is there a list of APIs we can refer to?

### LINKED ISSUE #67: Closing the documentation gap
Hey y'all, have implemented this a couple of times, but not often enough that I remember everything off the top of my head. Thought I'd try to contribute a couple code snippets that I use for reference to help others who are looking to use this module. Wasn't sure if this would warrant a pull request or not. If I'm dropping this in the wrong spot my apologies.

After installing the module and adding a config in `nuxt.config.js` following the [existing docs](https://github.com/nuxt-community/recaptcha-module), I set up a serverMiddleware to handle sending the recaptcha token along to the recaptcha service with the secret sitekey, without exposing it to the client (though I've ended up exposing my sitekey in commits more than once, ha).

In `nuxt.config.js`:
```
  serverMiddleware: ['~/api/recaptchaAuth'],

  ...

    modules: [ '@nuxtjs/recaptcha'],

  ...

  recaptcha: {
    hideBadge: true, // Hide badge element (v3 & v2 via size=invisible)
    siteKey: 'YOUR_SITEKEY', // Site key for requests
    version: 3, // Version, I like 3 because it's invisible 👾 
  },
```

In `~/api/recaptchaAuth`:
```
const app = require('express')()
const cors = require('cors')
const bodyParser = require('body-parser')
import axios from 'axios' // I want to use @nuxtjs/axios here but not savvy enough to figure out how to import 🤷‍♂️ 

const SECRET_KEY = 'YOUR_SECRET_KEY_HERE'

app.use([bodyParser.json(), cors()])
// I allowed cors here so I could test from 127.0.0.1 
// in addition to localhost, but you may want to remove 
// this before publishing your app/site.
app.options('*', cors()) 

app.post('/recaptcha', async (req, res) => {
  try {
    const { host } = req.headers
    const { token } = req.body

    if (!token) {
      res.status(400).send('No token!')
      throw 'No token!'
    }

    const { data } = await axios.post(
      `https://www.google.com/recaptcha/api/siteverify?secret=${SECRET_KEY}&response=${token}`,
    )

    res.send(data)
  } catch (e) {
    console.log('ReCaptcha error:', e)
    res.status(500).send(`<p>ReCaptcha error:</p> <pre>${e}</pre>`)
  }
})

module.exports = app
```

In the component/page where you want to use recaptcha, add a method (in this example, `~/components/nav.vue`, to hide and show contact info):
```
  methods: {
    async show() {
      try {
        const token = await this.$recaptcha.execute('contact') //my action is named contact but this can be whatever
        const res = await this.$axios.$post('/recaptcha', { token })

        if (res.success === true) this.$modal.show('contact-modal')
        else throw new Error('looks like you might not be a human :(')
      } catch (e) {
        console.log('Error:', e)
      }
    },
  },
```

And to complete my V3 example, I need to include the disclaimer:
```
  <p class="disclaimer">
    This site is protected by reCAPTCHA and the Google
    <a href="https://policies.google.com/privacy">Privacy Policy</a>
    and
    <a href="https://policies.google.com/terms">Terms of Service</a>
    apply.
  </p>
```

### PR DESCRIPTION
**Pull request by farnabaz**
- [X] Add module usage
- [x] Add server-site validation example

close #42
close #64 
close #67

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +58,68 @@ using top level options
 }
 ```
 
+## Usage
+
+### reCAPTCHA v2
+1. Add `<recaptcha>` component inside your form:
```
Comment (Atinux) [inline README.md:64]:
```suggestion

1. Add `<recaptcha>` component inside your form:
```

File: README.md [hunk 2]
```diff
@@ -58,6 +58,69 @@ using top level options
...
+    await this.$recaptcha.reset()
+  } catch (error) {
+    console.log('Login error:', error)
+  }
+},
+```
+See: [v2 example](https://github.com/nuxt-community/recaptcha-module/tree/master/example/v2)
+
+### reCAPTCHA v3
+1. Call `init` function inside `mounted` hook of your page
```
Comment (Atinux) [inline README.md:93]:
```suggestion

1. Call `init` function inside `mounted` hook of your page
```

File: README.md [hunk 3]
```diff
@@ -58,6 +58,69 @@ using top level options
...
+```js
+async mounted() {
+  try {
+    await this.$recaptcha.init()
+  } catch (e) {
+    console.error(e);
+  }
+}
+```
+2. Call `execute` function form submit handler to get reCAPTCHA token:
```
Comment (Atinux) [inline README.md:103]:
```suggestion

2. Call `execute` function form submit handler to get reCAPTCHA token:
```

File: README.md [hunk 4]
```diff
@@ -58,6 +58,69 @@ using top level options
...
+    console.log('ReCaptcha token:', token)
+
+    // send token to server alongside your form data
+
+  } catch (error) {
+    console.log('Login error:', error)
+  }
+}
+```
+See: [v3 example](https://github.com/nuxt-community/recaptcha-module/tree/master/example/v3)
```
Comment (Atinux) [inline README.md:117]:
```suggestion

See: [v3 example](https://github.com/nuxt-community/recaptcha-module/tree/master/example/v3)
```

File: README.md [hunk 5]
```diff
@@ -58,6 +58,69 @@ using top level options
...
+
+### Server Side
```
Comment (Atinux) [inline README.md:119]:
```suggestion

### Server Side
```

File: README.md [hunk 6]
```diff
@@ -58,6 +58,69 @@ using top level options
...
+1. Add `<recaptcha>` component inside your form:
+```vue
+<form @submit.prevent="onSubmit">
+  <input autocomplete="true" placeholder="Email" type="email" v-model="email">
+  <input autocomplete="current-password" placeholder="Password" type="password" v-model="password">
+  <recaptcha />
+  <button type="submit">Sign In</button>
+</form>
+```
+2. Call `getResponse` inside form submit handler to get reCAPTCHA token:
```
Comment (Atinux) [inline README.md:74]:
```suggestion

2. Call `getResponse` inside form submit handler to get reCAPTCHA token:
```

File: README.md [hunk 7]
```diff
@@ -58,6 +58,69 @@ using top level options
 }
 ```
 
+## Usage
+
+### reCAPTCHA v2
+
+1. Add `<recaptcha>` component inside your form:
+```vue
```
Comment (Atinux) [inline README.md:66]:
```suggestion

```vue
```

File: package.json [hunk 8]
```diff
@@ -39,6 +39,7 @@
     "@babel/preset-env": "latest",
     "@commitlint/cli": "latest",
     "@commitlint/config-conventional": "latest",
+    "@nuxt/h2": "latest",
```
Comment (Atinux) [inline package.json:42]: I believe we should use `h3` now: https://github.com/nuxt-contrib/h3

File: example/v2/api/recaptcha.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import { useBody } from '@nuxt/h2'
+import { $fetch } from 'ohmyfetch/node'
+
+const SECRET_KEY = '6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe' // Place your secret key here
```
Comment (Atinux) [inline example/v2/api/recaptcha.js:4]: Shall we also recommend to use `process.env.RECAPTCHA_SECRET`?

File: package.json (hunk 8 above)
Comment (farnabaz) [reply package.json:42]: Do you suggest to create servers?  
I just created a middleware function to show the concept

Comment (Atinux) [reply package.json:42]: I like the middleware, it's just that @nuxt/h2 is now `h3` :D

Comment (farnabaz) [reply package.json:42]: Oh, I didn't know 😄

**Pull request merged**